var/spack/repos/builtin/packages/openblas/package.py

```diff
@@ -184,6 +184,17 @@ class Openblas(MakefilePackage):
         when="@0.3.21 %gcc@:9",
     )
 
+    # Some installations of clang and libomp have non-standard locations for
+    # libomp. OpenBLAS adds the correct linker flags but overwrites the
+    # variables in a couple places, causing link-time failures.
+    patch("openblas_append_lflags.patch", when="threads=openmp %clang")
```

Comment:
You have to restrict to the versions where it's necessary.

https://github.com/OpenMathLib/OpenBLAS/commit/9a0e9c8b69ca1441ac7949a73de237cbff95812e

From where does it apply? Looks like it's in  v0.3.24. Probably fine to apply the patch for all compilers.

Reply:
Yep. Meant to update it when the patch actually landed upstream but didn't get around to it. Fixed now.